### ISSUE TITLE
Memory leaks in Phase 2 muon code

### ISSUE THREAD

**Issue by kpedro88** [cms-sw/cmssw#20772]
@slava77 identified a few memory leaks in the phase 2 muon code:
[RecoMuon/MuonIdentification/plugins/ME0SegmentMatcher.cc](https://github.com/cms-sw/cmssw/blob/09c3fce6626f70fd04223e7dacebf0b485f73f54/RecoMuon/MuonIdentification/plugins/ME0SegmentMatcher.cc#L172)
```
Plane *plane = new Plane(Surface::PositionType(0,0,zValue),Surface::RotationType());
```

[RecoLocalMuon/GEMSegment/plugins/GEMSegmentBuilder.cc](https://github.com/cms-sw/cmssw/blob/09c3fce6626f70fd04223e7dacebf0b485f73f54/RecoLocalMuon/GEMSegment/plugins/GEMSegmentBuilder.cc#L52)
```
ensembleRH[id.rawId()].push_back(it2->clone());
```
([`GEMRecHit::Clone()`](https://github.com/cms-sw/cmssw/blob/01dc8551b318859665cde533637fe7ae438e7aea/DataFormats/GEMRecHit/src/GEMRecHit.cc#L65) returns `new`)

These leaks are minor but should be addressed.

attn: @jshlee

**Comment by kpedro88** [comment]
assign reco,upgrade

**Comment by slava77** [comment]
assign reconstruction

**Comment by slava77** [comment]
@jshlee 
please clarify on the status of following up on the memory leaks reported here
Thank you.

**Comment by jshlee** [comment]
@slava77 - yes will do. wasn't sure who it was assigned to.

**Comment by kpedro88** [comment]
+1

**Comment by slava77** [comment]
+1
fixed in #21146
